### DISCUSSION TITLE
Menu Entry Swapper: Anti-venom and Anti-venom+ pots share mappings?

### DISCUSSION THREAD

**Discussion by MarkHarms** [runelite/runelite#16193] [Q&A] [answered]
Menu entry swapper, in all the forms I know of, usually handles entry swapping by using generic ids (this may be the wrong word), for example, setting left click on Super Attack(1) Potion will set the same setting for 2, 3, and 4 dose potions. This is intentional and expected.

However, I noticed a case where I don't believe this should be happening, this is between the Anti-venom and Anti-venom+ potions. They are entirely distinct items so I don't believe they should map to the same swaps. Are these 'generic' ids something that RuneLite maps and can be fixed, or is this simply something that comes from the OSRS code itself that cannot be fixed?

I also do not have the Potions to test for myself, but I am curious if the same thing happens for Antidote+ and Antidote++ potions.

**Comment by geheur** [comment]
Runelite maps these based on the item's name (https://github.com/abextm/runelite-cache-code-updater/blob/master/src/main/java/net/runelite/cache/codeupdater/apifiles/ItemVariationsUpdate.java iirc). It ignores many bits of text including "+". It's probably not easy to change these categories without messing up people's existing menu swaps and bank tags.

**Comment by Nightfirecat** [comment, marked as answer]
We should probably see if we can make it not categorize Anti-venom and Anti-venom+ as the same item's variation

**Reply by Nightfirecat** [reply]
This has been fixed in our cache code updater as of v1.1.11 (included via commit abextm/runelite-cache-code-updater@9086bfad2e54dd10375cf71f9d7904fb4a2e009f), thus should be working as you expect for anti-venom and antidote potions as of our next release.

**Reply by MarkHarms** [reply]
Thank you!
